docs/user-guide/configure.md

```diff
@@ -254,6 +254,119 @@ But the following pattern would be reported as a warning:
 a[data-auto="1"] {}
 ```
 
+## `languageOptions`
+
+You can customize the syntax to define or extend the syntax for at-rules, properties, types, and CSS-wide keywords.
+
+### `syntax`
+
+You can extend or modify the default CSS syntax to customize the following aspects:
+
+- `atRules`: Define custom at-rules with specific `prelude` and `descriptors` syntax
+- `cssWideKeywords`: Extend the list of CSS-wide keywords with custom values
+- `properties`: Customize the syntax of specific properties
+- `types`: Extend or modify type definitions used in property values
+
+```json
+{
+  "languageOptions": {
+    "syntax": {
+      "atRules": {
+        "example": {
+          "comment": "Example at-rule",
+          "prelude": "<custom-ident>",
+          "descriptors": {
+            "foo": "<number>",
+            "bar": "<color>"
+          }
+        }
+      },
+      "cssWideKeywords": ["my-global-value"],
+      "properties": { "top": "| <--foo()>" },
+      "types": { "--foo()": "--foo( <length-percentage> )" }
+    }
+  },
+  "rules": {
+    "at-rule-descriptor-no-unknown": true,
+    "at-rule-descriptor-value-no-unknown": true,
+    "at-rule-prelude-no-invalid": true,
+    "declaration-property-value-no-unknown": true
+  }
+}
+```
+
+The following rules are configured via the `languageOptions` property:
+
+- [`at-rule-descriptor-no-unknown`](../../lib/rules/at-rule-descriptor-no-unknown/README.md)
+- [`at-rule-descriptor-value-no-unknown`](../../lib/rules/at-rule-descriptor-value-no-unknown/README.md)
+- [`at-rule-prelude-no-invalid`](../../lib/rules/at-rule-prelude-no-invalid/README.md)
+<!-- TODO: Add 'declaration-property-value-no-unknown' after options are deprecated -->
+
+#### `AtRules`
```

Comment:
Would it be better to provide more detailed explanations for each option?

Reply:
Let's add some common examples to each option, e.g. supporting a new function for `<color>` type.

Reply:
I've added explanations to convey their connection to the rules.
[Add more examples](https://github.com/stylelint/stylelint/pull/8297/commits/57ff9a3fb45587ededf7d495e85984ff841f4b16)